doc/integrating_applications/topics/adding_dropbox_connection_middle.adoc

```diff
@@ -0,0 +1,36 @@
+[id='adding-dropbox-connection-middle']
+= Adding a Dropbox connection as a middle connection
+
+In an integration, to add a Dropbox connection between the start and 
+finish connections:
+
+. Add the start and finish connections to the integration.
+. In the left panel, hover over the plus sign that is in the location
+where you want to add a Dropbox connection.
+. In the popup, click *Add a Connection*.
+. On the *Choose a Connection* page, click the Dropbox connection that you 
+want the integration to use. 
+
+. On the *Choose an Action* page, select one of the following actions:
++
+* *Upload a file to Dropbox* to add the current integration data to the
+Dropbox account that this connection accesses. To configure this
+action:
+.. In the *Remote Path* field, enter the path of the Dropbox folder 
+in which to store the integration data. 
+
+.. For the *Upload mode* select:
++
+** *Add* to rename a file that you are uploading if a file with the
```

Comment:
In this case if the file has the same name of the file in the dropbox path, the upload won't be completed and the existing won't be overwritten.

Reply:
Will the integration then be in an error state?
If not, how does the user know that the upload was not completed?
What should the user do to correct the situation?
(I can't believe how fast you read!)

Reply:
The upload will complete without error, but the file won't be overwritten. No error.

Add option is from Dropbox API stuff and it will ensure you won't overwrite files while uploading, while with force you will overwrite existing files.

Reply:
Sorry, I am still not clear about the behavior. It sounds like if the user selects Add for the Upload Mode and there is a file already in Dropbox with the same name then the file being uploaded is not Added. I must be wrong about that. Hmmmm. Is this the behavior:
- The file that is already in Dropbox is not overwritten.
- The file being uploaded is somehow renamed - how?
- There is no error, the upload operation completes, the integration continues execution. 
Or is this the behavior:
- The file that is already in Dropbox is not overwritten.
- The file that is being uploaded is lost.
- There is no error and the integration continues. 
Thanks for your patience!

Reply:
If the file uploaded is not lost, then happens to it?

Reply:
Suppose you're trying to get a file from an FTP as starting connection and you want to upload to dropbox, in the next poll you'll get the same file and trying to upload it to dropbox, so it's not lost, there will be a sequence of uploading operation, that won't complete because you're using the add mode. We can stress this in the docs and make the users aware that using add mode won't ensure that the file will be uploaded.

Reply:
Okay. That helps. When you say "same file", does that mean the same content?
Should we be telling this to the user: 
Select Add when you want to upload the file only when its content has been updated.
Select Force when you want to ensure that the file is uploaded even if it has the same content as the file that is already in place.
--
I seem to be diverging from the Camel user doc, which says: "in case of add the new file will be renamed if a file with the same name already exists on dropbox. "

Reply:
I mean same name. If you try to upload a file with the same name of a file in dropbox remote path, then it won't be uploaded, the content can be different. This morning I tested both Add mode and Force mode in Syndesis and when I upload a file with the same name of a file specified in the dropbox remote path, the upload hasn't been completed, while using force mode the upload has been completed.

Reply:
My opinion is:

- Select Add when you want to upload the file only when it's not already present in the same Dropbox path.
- Select Force when you want to ensure that the file is uploaded even if it is already present in the same Dropbox path.

Reply:
Perfect :-)

Reply:
Really happy dance for this one  !